tests/types/test_numpy.py

```diff
@@ -0,0 +1,379 @@
+        (32_767, "'32767'::int2"),
+        (0, "'0'::int2"),
+        (45, "'45'::int2"),
+    ],
+)
+@pytest.mark.parametrize("fmt_in", PyFormat)
+def test_dump_numpy_short(numpy_conn, val, expr, fmt_in):
+
+    val = np.short(val)
+
+    assert isinstance(val, np.short)
+    assert np.short is np.int16
+
+    cur = numpy_conn.cursor()
+
+    cur.execute(f"select pg_typeof({expr}) = pg_typeof(%{fmt_in.value})", (val,))
+    cur.fetchone()[0] is True
+
+    cur.execute(f"select {expr} = %{fmt_in.value}", (val,))
+    assert cur.fetchone()[0] is True
+
+
+@pytest.mark.parametrize(
+    "val, expr",
+    [
+        (-2_147_483_648, "'-2147483648'::int4"),
+        (2_147_483_647, "'2147483647'::int4"),
+        (0, "'0'::int4"),
+        (45, "'45'::int4"),
+    ],
+)
+@pytest.mark.parametrize("fmt_in", PyFormat)
+def test_dump_numpy_intc(numpy_conn, val, expr, fmt_in):
+
+    val = np.intc(val)
+
+    assert isinstance(val, np.intc)
+    assert np.intc is np.int32
+
+    cur = numpy_conn.cursor()
+
+    cur.execute(f"select pg_typeof({expr}) = pg_typeof(%{fmt_in.value})", (val,))
+    cur.fetchone()[0] is True
+
+    cur.execute(f"select {expr} = %{fmt_in.value}", (val,))
+    assert cur.fetchone()[0] is True
+
+
+@pytest.mark.parametrize(
+    "val, expr",
+    [
+        (-9_223_372_036_854_775_808, "'-9223372036854775808'::int8"),
+        (9_223_372_036_854_775_807, "'9223372036854775807'::int8"),
+        (0, "'0'::int8"),
+        (45, "'45'::int8"),
+    ],
+)
+@pytest.mark.parametrize("fmt_in", PyFormat)
+def test_dump_numpy_int_(numpy_conn, val, expr, fmt_in):
+
+    val = np.int_(val)
+
+    assert isinstance(val, np.int_)
+    assert np.int_ is np.int64
+
+    cur = numpy_conn.cursor()
+
+    cur.execute(f"select pg_typeof({expr}) = pg_typeof(%{fmt_in.value})", (val,))
+    cur.fetchone()[0] is True
+
+    cur.execute(f"select {expr} = %{fmt_in.value}", (val,))
+    assert cur.fetchone()[0] is True
+
+
+@pytest.mark.parametrize(
+    "val", [(9_223_372_036_854_775_808,), (-9_223_372_036_854_775_809,)]
+)
+def test_dump_python_overflow(val):
+    with pytest.raises(OverflowError) as excinfo:
+        np.int_(val)
+
+    assert excinfo
+
+
+@pytest.mark.parametrize(
+    "val, expr",
+    [
+        (True, "'t'::bool"),
+        (False, "'f'::bool"),
+    ],
+)
+@pytest.mark.parametrize("fmt_in", PyFormat)
+def test_dump_numpy_bool_(numpy_conn, val, expr, fmt_in):
+
+    val = np.bool_(val)
+
+    assert isinstance(val, np.bool_)
+    assert np.bool_ is np.bool8
+
+    cur = numpy_conn.cursor()
+
+    cur.execute(f"select pg_typeof({expr}) = pg_typeof(%{fmt_in.value})", (val,))
+    cur.fetchone()[0] is True
+
+    cur.execute(f"select {expr} = %{fmt_in.value}", (bool(val),))
+    assert cur.fetchone()[0] is True
+
+
+@pytest.mark.parametrize(
+    "val, expr",
+    [(0, "'0'::int2"), (255, "'255'::int2")],
+)
+@pytest.mark.parametrize("fmt_in", PyFormat)
+def test_dump_numpy_ubyte_(numpy_conn, val, expr, fmt_in):
```

Comment:
Please call the tests after the canonical types, not the aliases. You can also avoid `numpy` in the name, as it's clear from the test module.

```suggestion
def test_dump_uint8(numpy_conn, val, expr, fmt_in):
```